### ISSUE TITLE
Wrong certificate is chosen when using `TLSStore` with multiple certificates.

### ISSUE THREAD

**Issue by reitermarkus** [traefik/traefik#9198]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I have this `TLSStore` configured in my K8s cluster:

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: TLSStore
metadata:
  name: default
  namespace: traefik-system
spec:
  certificates:
    - secretName: example.org
    - secretName: example.com
  defaultCertificate:
    secretName: example.org
```

Both secrets contain wildcard certificates for the respective domains.

I also have this `IngressRoute` deployed:

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: app
  namespace: default
spec:
  entryPoints:
    - websecure
  routes:
    - kind: Rule
      match: Host(`app.example.com`)
      services:
        - name: app
          namespace: default
          port: http
  tls:
    store:
      name: default
```

When I access `https://app.example.com`, I would expect Traefik to choose the matching `example.com` certificate.

### What did you see instead?

When I access `https://app.example.com`, the default `example.org` certificate is chosen.

I expected Traefik to choose the matching `example.com` wildcard certificate.

### What version of Traefik are you using?

2.8.0

### What is your environment & configuration?

Helm values:

```yaml
additionalArguments:
  - --serverstransport.insecureskipverify=true # Don't validate certificates for backends.
  - --providers.kubernetescrd.allowCrossNamespace=true # Allow accessing middlewares across namespaces.
```

### If applicable, please paste the log output in DEBUG level

```
time="2022-07-15T21:32:19Z" level=debug msg="Serving default certificate for request: \"app.example.us\""
```

**Comment by rtribotte** [comment]
Hello @reitermarkus,

Thanks for your interest in Traefik!

The issue seems to be related to an empty or wrong SNI.
Can you share/explain how the client request is made?

To make sure that the client request doesn't contain any SNI or the valid SNI, you can enable temporarily the [SNI strict](https://doc.traefik.io/traefik/master/https/tls/#strict-sni-checking) option.

**Comment by reitermarkus** [comment]
> Can you share/explain how the client request is made?

The request is made using a browser, i.e. visiting the URL.

> you can enable the [SNI strict](https://doc.traefik.io/traefik/master/https/tls/#strict-sni-checking) option temporarily

Okay, I deployed

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: TLSOption
metadata:
  name: default
  namespace: traefik-system
spec:
  sniStrict: true
```

and it seems to be related to empty or wrong SNI. 

With strict SNI, I am getting an “unrecognized name” error, with Chromium, Safari and `curl`.

**Comment by rtribotte** [comment]
Hello @reitermarkus,

So maybe the server name is empty, but it can be checked in the debug log that, in your case, would look like: `TLS: strict SNI enabled - No certificate found for domain: "app.example.com", closing connection`. If it contains the correct domain (server name), then it means that there was no matching certificate found in the `TLSStore`.

Nonetheless, this seems to be a configuration issue and GitHub issue tracker is dedicated to bug and feature requests.
For help on your configuration, please join our [Community Forum](https://community.traefik.io/) and reach out to us on the [Traefik section](https://community.traefik.io/c/traefik). 

We will close this issue accordingly but feel free to re-open it if you think that we missed something.

**Comment by reitermarkus** [comment]
Thanks, @rtribotte, I will join the Community Forum.

I did in fact get the

```
TLS: strict SNI enabled - No certificate found for domain: "app.example.com", closing connection
```

error after enabling debug logging again.

**Comment by reitermarkus** [comment]
@rtribotte, I found the issue.

Since this is not a fresh cluster, the Traefik CRDs were not up-to-date. Apparently upgrading CRDs is not supported via `helm`, which is why they were out of date.

Specifically, the `certificates` key in the `TLSStore` was not actually present in the CRD, and apparently the Ansible `kubernetes.core.k8s` module silently swallows the error which would be shown when running `kubectl apply` manually:

```
error: error validating "traefik/tlsstore.yml": error validating data: ValidationError(TLSStore.spec): unknown field "certificates" in us.containo.traefik.v1alpha1.TLSStore.spec; if you choose to ignore these errors, turn validation off with --validate=false
```

Long story short, the updated `TLSStore` was never actually deployed and only ever contained `defaultCertificate`.
